### PR TITLE
Use only single rollup-plugin-copy() to avoid race condition

### PR DESCRIPTION
**Pull request by JeffersGlass**
#839 improved rollup logic - it always copies `build` folder to `examples/build`, regardless of whether this is a production build or not. However, it seems that `rollup-plugin-copy` doesn't behave nicely when used multiple times in a single rollup config. Possibly a race condition or data corruption? Errors include `EONENT: file not found 'build/index.html'`, `EBUSY: resource busy or locked` on Linux and Windows build systems.

This PR moves that logic outside of the exported rollup config, to avoid the error.

### CODE REVIEW

File: pyscriptjs/rollup.config.js [hunk 1]
```diff
@@ -47,14 +61,7 @@ export default {
     }),
     // This will make sure that examples will always get the latest build folder
     !production && copy({
```
Comment (tedpatrick) [inline pyscriptjs/rollup.config.js:63]: If we are managing file copy paths, let's remove the `!production` on line 63 limiting this to not-production

File: pyscriptjs/rollup.config.js [hunk 2]
```diff
@@ -47,14 +61,7 @@ export default {
...
-      ]
-    }),
-    copy({
-      targets: [
-        { src: 'public/index.html', dest: 'build' },
-      ]
+      targets: copy_targets
     }),
     !production && serve(),
     !production && livereload("public"),
```
Comment (tedpatrick) [inline pyscriptjs/rollup.config.js:67]: These should be deleted as they are duplicate and the directory listing does not need to be livereload
```js
!production && serve(),
!production && livereload("public"),
```

**Comment by tedpatrick** [comment]
I think we need to revisit this one. 

1. Definition of `production` is polluted by https://github.com/pyscript/pyscript/blob/main/pyscriptjs/rollup.config.js#L12
2. Copying the directory listing (public/index.html) needs to run on all builds, not just `!production`
3. Some duplicate rollup commands to remove https://github.com/pyscript/pyscript/blob/main/pyscriptjs/rollup.config.js#L59-L60

File: pyscriptjs/rollup.config.js (hunk 1 above)
Comment (JeffersGlass) [reply pyscriptjs/rollup.config.js:63]: Agreed and removed.

File: pyscriptjs/rollup.config.js (hunk 2 above)
Comment (JeffersGlass) [reply pyscriptjs/rollup.config.js:67]: Agreed and removed.

**Comment by JeffersGlass** [comment]
Thanks @tedpatrick ! I've adjusted the file per your thoughts, removing the duplicate commands and cleaning up the definition of `copy_targets` at the top there.

**Comment by tedpatrick** [comment]
It looks like the serve command is bing called in github actions and stops and listens
Screen Shot 2022-10-11 at 9 11 41 PM

**Comment by tedpatrick** [comment]
I think it is this line defining production
```js
const production = !process.env.ROLLUP_WATCH || (process.env.NODE_ENV === "production");
```

**Comment by JeffersGlass** [comment]
> I think it is this line defining `production`
> 
> ```js
> const production = !process.env.ROLLUP_WATCH || (process.env.NODE_ENV === "production");
> ```

Thanks for stopping the action - good catch! And interesting - that line has been in that form for ages... but in the commit that triggered that broken build 06fe46a, I had changed it to:
```diff
-   const production = !process.env.ROLLUP_WATCH || (process.env.NODE_ENV === "production");
+   const production = (process.env.NODE_ENV === "production");
```
Maybe that ROLLUP_WATCH flag is doing more than I thought... let me push a commit adding that flag back in and we'll see what happens in CI...

**Comment by tedpatrick** [comment]
Rerunning actions. It looked to fail in an integration test. But the fact that it reaches integration is positive!

**Comment by tedpatrick** [comment]
Yes I agree, the `!process.env.ROLLUP_WATCH` is confusing

**Review by tedpatrick — APPROVED** [review]

**Comment by tedpatrick** [comment]
@JeffersGlass Merge when ready. 

Merge policy: You build it, you merge it!

**Comment by tedpatrick** [comment]
Great work @JeffersGlass, Thanks.

**Pull request merged**